agent/consul/state/catalog_events.go

```diff
@@ -358,6 +361,10 @@ func ServiceHealthEventsFromChanges(tx ReadTxn, changes Changes) ([]stream.Event
 	}
 	events = append(events, connectEvents...)
 
+	// Duplicate all of these into the preferconnect topic
```

Comment:
In the common case I would expect folks wouldn't use `?prefer-connect` so would this event duplication be too wasteful?

Reply:
Change events are emitted even when there are no subscribers. But when there are no subscribers the events will effectively be GC'ed immediately, so I think worst case scenario is extra unnecessary time spend GC'ing.